server/pkg/project/project.go

```diff
@@ -119,6 +129,20 @@ func (p *Project) UpdateLicense(license string) {
 	p.license = license
 }
 
+func (p *Project) SetStarCount(count int64) {
+	if p == nil {
+		return
+	}
+	p.starCount = count
+}
+
+func (p *Project) SetStarredBy(starredBy []string) {
+	if p == nil {
+		return
+	}
+	p.starredBy = starredBy
+}
+
```

Comment:
```suggestion
func (p *Project) Star(uId user.Id) {
	if p == nil {
		return
	}
	p.starCount ++
	p.starredBy = append(p.strredBy, uId)
}

func (p *Project) Unstar(uId user.Id) {
	...
}
```
It's better to do it this way to keep the project in a correct state